demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -278,24 +305,44 @@ static mbedtls_sha256_context hashContext = { 0 };
 /**
  *  @brief Configurations of the AWS credentials sent to sigV4 library for generating the Authorization Header.
  */
-SigV4Credentials_t sigvCreds = { 0 };
+static SigV4Credentials_t sigvCreds = { 0 };
 
 /**
  * @brief Represents a response returned from an AWS IOT credential provider.
  */
-HTTPResponse_t credentialResponse = { 0 };
+static HTTPResponse_t credentialResponse = { 0 };
 
 /**
  * @brief Buffer used in the demo for storing temporary credentials
  * received from AWS TOT credential provider.
  */
-uint8_t pAwsIotHttpBuffer[ CREDENTIAL_BUFFER_LENGTH ] = { 0 };
+static uint8_t pAwsIotHttpBuffer[ CREDENTIAL_BUFFER_LENGTH ] = { 0 };
 
 /**
  * @brief Represents date in ISO8601 format used in the HTTP requests sent to AWS S3.
  */
 static char pDateISO8601[ SIGV4_ISO_STRING_LEN ] = { 0 };
 
+/**
+ * @brief Represents hash digest of payload.
+ */
+static char pPayloadHashDigest[ SHA256_HASH_DIGEST_LENGTH ];
+
+/**
+ * @brief Represents hex encoded hash digest of payload.
+ */
+static char hexencoded[ HEX_ENCODED_SHA256_HASH_DIGEST_LENGTH ];
+
+/**
+ * @brief Represents Authorization header value generated using SigV4 library.
+ */
+static char pSigv4Auth[ AWS_HTTP_AUTH_HEADER_VALUE_LEN ];
+
+/**
+ * @brief Represents Length of Authorization header value generated using SigV4 library.
+ */
+static size_t sigv4AuthLen = AWS_HTTP_AUTH_HEADER_VALUE_LEN;
```

Comment:
If there isn't a need of keeping this variable global, this can be moved to the stack

Reply:
It is used across multiple function so I made it global.